### PR TITLE
broker and flux-comms cleanup

### PR DESCRIPTION
**Pull request by garlick**
This is one of several broker cleanup peeled off of work towards distributed instance startup.

This one doesn't change a lot materially:
* turn `flux-comms` into a builtin
* use optparse in `flux-comms`
* rename `flux comms idle` to `flux comms lspeer` (to match RPC)
* move lspeer handler from broker main to `broker/overlay.c`
* general cleanup in `overlay.c`

### CODE REVIEW

File: src/cmd/builtin/comms.c [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    log_init ("flux-comms");
+
+    if (optparse_run_subcommand (p, ac, av) != OPTPARSE_SUCCESS)
+        exit (1);
+    return (0);
+}
+
+static struct optparse_subcommand comms_subcmds[] = {
+    { "lspeer",
+      "[OPTIONS]",
```
Comment (grondo) [inline src/cmd/builtin/comms.c:100]: Very minor, but `flux-comms` subcommands don't take any `[OPTIONS]`, so perhaps the usage string here and below should be NULL or empty? (Though I guess they all do take `-h, --help`, so perhaps it is fine.

**Comment by garlick** [comment]
Just forced a push with two things fixed.  Thanks!

File: src/broker/overlay.c [hunk 2]
```diff
@@ -593,6 +564,54 @@ void overlay_set_monitor_cb (struct overlay *ov,
     ov->child_monitor_arg = arg;
 }
 
+static json_t *list_object_create (struct overlay *ov)
```
Comment (chu11) [inline src/broker/overlay.c:567]: whenever I see a function starting with `list_` I think of a singly linked list.  Perhaps rename function?

**Comment by garlick** [comment]
Force pushed after renaming that function to `lspeer_object_create()`.

**Comment by grondo** [comment]
Went ahead and set MWP label. Thanks!

**Comment by garlick** [comment]
Thanks!

**Pull request merged**